### PR TITLE
Annotations for defaultunits.lua

### PR DESCRIPTION
**Pull request by MrRowey**
Annotated most of the file any that I was unsure of have left as Unknown

### CODE REVIEW

File: lua/defaultunits.lua [hunk 1]
```diff
@@ -288,6 +312,9 @@ StructureUnit = Class(Unit) {
     end,
 
     -- Modified to use same upgrade logic as the ui. This adds more upgrade options via General.UpgradesFromBase blueprint option
+    ---@param self StructureUnit
+    ---@param unitBeingBuilt StructureUnit
```
Comment (Garanas) [inline lua/defaultunits.lua:316]: Doesn't have to be a structure, a factory produces units

File: lua/defaultunits.lua [hunk 2]
```diff
@@ -395,6 +422,9 @@ StructureUnit = Class(Unit) {
         end,
     },
 
+    ---@param self StructureUnit
+    ---@param builder Builder
```
Comment (Garanas) [inline lua/defaultunits.lua:426]: Better to have `Unit` instead of `Builder`

File: lua/defaultunits.lua [hunk 3]
```diff
@@ -427,26 +460,41 @@ StructureUnit = Class(Unit) {
         Unit.StopBeingBuiltEffects(self, builder, layer)
     end,
 
+    ---comment
+    ---@param self StructureUnit
+    ---@param unitBeingBuilt StructureUnit
```
Comment (Garanas) [inline lua/defaultunits.lua:465]: Can also be a regular `Unit`

Comment (Garanas) [reply lua/defaultunits.lua:465]: Same for all the others

File: lua/defaultunits.lua [hunk 4]
```diff
@@ -465,6 +513,8 @@ StructureUnit = Class(Unit) {
         self:CreateTarmac(true, true, true, orient, currentBP, currentBP.DeathLifetime or 300)
     end,
 
+    ---@param self StructureUnit
+    ---@param wreckage Wreckage
```
Comment (Garanas) [inline lua/defaultunits.lua:517]: Better to use `Prop`

File: lua/defaultunits.lua [hunk 5]
```diff
@@ -803,6 +881,9 @@ FactoryUnit = Class(StructureUnit) {
         end
     end,
 
+    ---@param self FactoryUnit
+    ---@param unitBeingBuilt FactoryUnit
```
Comment (Garanas) [inline lua/defaultunits.lua:885]: Can also be a regular unit

File: lua/defaultunits.lua [hunk 6]
```diff
@@ -832,6 +916,10 @@ FactoryUnit = Class(StructureUnit) {
     end,
 
     --- Adds a pause between unit productions
+    ---@param self FactoryUnit
+    ---@param productionpause FactoryUnit
```
Comment (Garanas) [inline lua/defaultunits.lua:920]: this is a number (it is used in the wait function)

File: lua/defaultunits.lua [hunk 7]
```diff
@@ -1844,6 +2079,9 @@ BaseTransport = Class() {
         unit:OnAttachedToTransport(self, attachBone)
     end,
 
+    ---@param self BaseTransport
+    ---@param attachBone string
```
Comment (Garanas) [inline lua/defaultunits.lua:2083]: can also be a number, there's a separate bone class / alias somewhere. @Hdt80bro do you know its name?

File: lua/defaultunits.lua [hunk 8]
```diff
@@ -1899,18 +2146,24 @@ BaseTransport = Class() {
 --- Base class for air transports.
 ---@class BaseTransport : AirUnit
 AirTransport = Class(AirUnit, BaseTransport) {
+
+    ---@param self BaseTransport
```
Comment (Garanas) [inline lua/defaultunits.lua:2150]: This should be an `AirTransport`, that inherits from both `AirUnit` and `BaseTransport`

**Comment by Garanas** [comment]
Best to use `any` if you're unsure, instead of `unknown`

File: lua/defaultunits.lua (hunk 7 above)
Comment (Hdt80bro) [reply lua/defaultunits.lua]: `Bone`

File: lua/defaultunits.lua [hunk 9]
```diff
@@ -1652,6 +1865,9 @@ AirUnit = Class(MobileUnit) {
         end
     end,
 
+    ---@param self AirUnit
+    ---@param new AirUnit
```
Comment (Garanas) [inline lua/defaultunits.lua:1869]: these are not air units, they are strings

File: lua/defaultunits.lua [hunk 10]
```diff
@@ -1759,6 +1988,10 @@ AirUnit = Class(MobileUnit) {
     end,
 
     --- Called when the unit is killed, but before it falls out of the sky and blows up.
+    ---@param self AirUnit
+    ---@param instigator AirUnit
```
Comment (Garanas) [inline lua/defaultunits.lua:1992]: instigator can be any unit

**Pull request merged**